### PR TITLE
[C-906] Migrate queue sagas and hls-utils to common

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Migrates queue sagas to common
Migrates hlsUtils to @audius/common

These are the final pieces to enable fully native audio

### CODE REVIEW

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-906-migrate-queue-hls-to-common

File: packages/web/src/common/store/queue/mobileSagas.ts [hunk 1]
```diff
@@ -72,7 +73,10 @@ function* getTrackInfo(id: ID, uid: UID) {
     : track.cover_art
 
   const m3u8Gateways = gateways.concat(PUBLIC_IPFS_GATEWAY)
-  const m3u8 = generateM3U8Variants(track.track_segments, [], m3u8Gateways)
+  const m3u8 = generateM3U8Variants({
```
Comment (raymondjacobson) [inline packages/web/src/common/store/queue/mobileSagas.ts:76]: don't we want to do this inside the component on the native side? or is that not necessary now?

**Review by raymondjacobson — APPROVED** [review]

File: packages/web/src/common/store/queue/mobileSagas.ts (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/common/store/queue/mobileSagas.ts:76]: yeah we do, but this is just for backwards compat on main!

Comment (raymondjacobson) [reply packages/web/src/common/store/queue/mobileSagas.ts:76]: ohhh i didn't recognize the diff base here. sounds great

**Review by sliptype — APPROVED** [review]
Awesome! thanks for converting to ts too

**Pull request merged**